packages/launch/src/hooks/use-plans.tsx

```diff
@@ -15,21 +15,35 @@ import LaunchContext from '../context';
 import { isPlanProduct } from '../utils';
 import type { PlanProductForFlow } from '../utils';
 
-export function usePlans(): {
+export function usePlans(
+	billingPeriod: Plans.PlanBillingPeriod = 'ANNUALLY'
+): {
 	defaultPaidPlan: Plans.Plan | undefined;
 	defaultFreePlan: Plans.Plan | undefined;
+	defaultFreePlanProduct: Plans.PlanProduct | undefined;
+	defaultPaidPlanProduct: Plans.PlanProduct | undefined;
 } {
 	const locale = useLocale();
 
-	const defaultPaidPlan = useSelect( ( select ) =>
-		select( PLANS_STORE ).getDefaultPaidPlan( locale )
-	);
+	return useSelect( ( select ) => {
+		const defaultFreePlan = select( PLANS_STORE ).getDefaultFreePlan( locale );
+		const defaultPaidPlan = select( PLANS_STORE ).getDefaultPaidPlan( locale );
+		const defaultPaidPlanProduct = select( PLANS_STORE ).getPlanProduct(
+			defaultPaidPlan?.periodAgnosticSlug,
+			billingPeriod
+		);
+		const defaultFreePlanProduct = select( PLANS_STORE ).getPlanProduct(
+			defaultFreePlan?.periodAgnosticSlug,
+			billingPeriod
+		);
```

Comment:
I think this is a bit cleaner and easier to maintain. What do you think?

```suggestion
		const plansStore = select(PLANS_STORE);
		const defaultFreePlan = plansStore.getDefaultFreePlan( locale );
		const defaultPaidPlan = plansStore.getDefaultPaidPlan( locale );
		const defaultPaidPlanProduct = plansStore.getPlanProduct(
			defaultPaidPlan?.periodAgnosticSlug,
			billingPeriod
		);
		const defaultFreePlanProduct = plansStore.getPlanProduct(
			defaultFreePlan?.periodAgnosticSlug,
			billingPeriod
		);
```

Reply:
Also, don't forget to add the second argument to `useSelect` function which in this case should be `[locale, billingPeriod]`. See p7H4VZ-2Q6-p2#comment-9214